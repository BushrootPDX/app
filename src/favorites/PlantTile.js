import React from 'react';

export default function PlantTile({ removeFromFaves, viewDetails, selectPlant, plant}) {
    let { name, image, _id} = plant;
    return (
        <div id={_id} onClick={() => selectPlant(_id)}>
            <img src={image} alt={name}/>
            <h6>{name}</h6>
            <button onClick={removeFromFaves}>remove</button>
            <button onClick={viewDetails}>view</button>
        </div>
    );

}