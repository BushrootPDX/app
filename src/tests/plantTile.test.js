import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PlantTile from '../favorites/PlantTile';

const removeFromFaves = () => 'Removed!';
const viewDetails = () => 'detailsViewed!';
const selectPlant = () => 'plant Slected!';
const plant = {name:'po',image:'image',_id:'123'};

describe('PlantTile component', () => {
    it('renders a plant tyle component with listening functions', () => {
        const wrapper = shallow(<PlantTile 
            removeFromFaves={removeFromFaves}
            viewDetails={viewDetails}
            selectPlant={selectPlant}
            plant={plant}
        />);
        expect(toJSON(wrapper)).toMatchSnapshot();

    });

});