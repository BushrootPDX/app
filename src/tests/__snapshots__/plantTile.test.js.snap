// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantTile component renders a plant tyle component with listening functions 1`] = `
<div
  id="123"
  onClick={[Function]}
>
  <img
    alt="po"
    src="image"
  />
  <h6>
    po
  </h6>
  <button
    onClick={[Function]}
  >
    remove
  </button>
  <button
    onClick={[Function]}
  >
    view
  </button>
</div>
`;
